server.js

```diff
@@ -21,9 +23,55 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  // res.send({responseMessage: "Hello Technigo!"}); // we are send when we would like to sent translate json object
+  res.json ({responseMessage: "Hello Technigo!"}) // it is sending as a json, which makes it more secure if there is some error
 });
 
+app.get("/library", (req, res) => {
+ const { title, isbn, authors } = req.query;
+   let books = booksData;
+
+  if (title) {
+    books = books.filter(singleBook => singleBook.title.toLowerCase() === title.toLowerCase()); // to be sure that API will read all possible way of writing it
+                                                                                                // it is important to use toLowerCase function
+  }
+
+  if (isbn){
+    books = books.filter(singleBook => singleBook.isbn == isbn);
+  }
+
+  if (authors) {
+    books = books.filter(singleBook => singleBook.authors.toLowerCase() === authors.toLowerCase())
+  }
+
+
+
+  res.status(200).json ({
+    success: true,
+    message: "OK",
+    body: {
+      booksData: books
+    }
+   }) // it is sending as a json, which makes it more secure if there is some error
+
+  
+
+});
+
+app.get("/library/:id", (req, res) => {
+  const singleBook = booksData.find((item) => {
+    return(item.bookID === + req.params.id)
+  });
+  console.log(singleBook)
+  res.json ({booksData  : singleBook}) // it is sending as a json, which makes it more secure if there is some error
+});
+
+app.get("/library", (req, res) => {
+  res.json (singleBook) // it is sending as a json, which makes it more secure if there is some error
+});
```

Comment:
Is this router needed ? It looks like it returning all data as in the first route at row 30. ☺️